### PR TITLE
Have sims test the eth_blockNumber endpoint

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

- Provide a description and a link to the underlying ticket

### What changes were made as part of this PR:

- Provide a high level list of the changes made
- List key areas for the reviewer 

### :rotating_light: Definition of done :rotating_light:
- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

File: go/obsclient/obsclient.go [hunk 1]
```diff
@@ -29,15 +26,3 @@ func NewObsClient(c rpc.Client) *ObsClient {
 func (oc *ObsClient) Close() {
 	oc.rpcClient.Stop()
 }
-
-// Blockchain Access
-
-// ChainID retrieves the current chain ID for transaction replay protection.
-func (oc *ObsClient) ChainID() (*big.Int, error) {
```
Comment (BedrockSquirrel) [inline go/obsclient/obsclient.go:36]: Why get rid of this?

File: integration/simulation/utils.go [hunk 2]
```diff
@@ -88,6 +90,23 @@ func getRollupHeader(client rpc.Client, hash gethcommon.Hash) *common.Header {
 	return result
 }
 
+// Uses the client to retrieve the current block number.
+func getBlockNumber(client rpc.Client) uint64 {
```
Comment (BedrockSquirrel) [inline integration/simulation/utils.go:94]: Can we add the method to Obsclient and remove this?

The analogous gethclient method looks like this:
```
// BlockNumber returns the most recent block number
func (ec *Client) BlockNumber(ctx context.Context) (uint64, error) {
	var result hexutil.Uint64
	err := ec.c.CallContext(ctx, &result, "eth_blockNumber")
	return uint64(result), err
}
```

Comment (BedrockSquirrel) [reply integration/simulation/utils.go:94]: Some of the other methods in here should be on the client too, could maybe leave a todo for them

**Review by BedrockSquirrel — CHANGES_REQUESTED** [review]
Generally LGTM but can we move the call to the unauthenticated client

File: go/obsclient/obsclient.go (hunk 1 above)
Comment (joeldudley) [reply go/obsclient/obsclient.go:36]: Unused. Seemed arbitrary to just have that one?

Comment (BedrockSquirrel) [reply go/obsclient/obsclient.go:36]: It's the client for unauthorised rpc requests., we just haven't added the rest yet.

Comment (joeldudley) [reply go/obsclient/obsclient.go:36]: Ok, I'll readd.

**Review by BedrockSquirrel — APPROVED** [review]
LGTM

**Pull request merged**